### PR TITLE
101 registation redux

### LINKED ISSUE #101: Write an action and reducer for registration
Write an action and a corresponding reducer for registration that can be called from a component.

Please see https://github.com/kowhai-2018/Final-Project/blob/development/client/actions/auth.js and https://github.com/kowhai-2018/Final-Project/blob/development/client/reducer/auth.js for example code.

Please note that in the example code the api and the action are in the same function but these should be separated. The api for handling the login will be in a separate api file, see issue #96.

### PR DESCRIPTION
**Pull request by benlittlenz**
closes #101 
Added Redux to project and also made action and reducer functions to handle the registration.
Also added an API call to retrieve data from backend and added tokens to work on local storage to deal with one level of indirection for authentication.

### CODE REVIEW

**Comment by ari-t** [comment]
This should use api #96 rather than having the api call in the action.

File: client/index.js [hunk 1]
```diff
@@ -1,14 +1,26 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
+import {applyMiddleware, compose, createStore} from 'redux'
 import {BrowserRouter as Router} from 'react-router-dom'
+import {Provider} from 'react-redux'
+import thunkMiddleware from 'redux-thunk'
 
 import App from './components/App'
+import reducer from './reducer'
```
Comment (emilyparkes) [inline client/index.js]: reducers please as it shall have more than one

File: client/reducer/auth.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import {getToken} from '../utils/tokens'
+
+const initialState = {
+  error: null,
+  loggedIn: !!getToken(), // Coerces object to boolean, so either returns true or false
```
Comment (emilyparkes) [inline client/reducer/auth.js]: no comments please

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: client/index.js (hunk 1 above)
Comment (benlittlenz) [reply client/index.js]: Made the changes

**Review by emilyparkes — CHANGES_REQUESTED** [review]
Please pull from dev now to make sure you don't have any conflicts.
Can't tell if it works without it. Please test that your actions and reducer behaves appropriately in redux dev tools.

**Comment by benlittlenz** [comment]
> Please pull from dev now to make sure you don't have any conflicts.
> Can't tell if it works without it. Please test that your actions and reducer behaves appropriately in redux dev tools.

Fixed conflics between dev branch and this one

**Review by emilyparkes — APPROVED** [review]

**Pull request merged**